src/particles/elements/Sol.H

```diff
@@ -0,0 +1,165 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Chad Mitchell, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_SOL_H
+#define IMPACTX_SOL_H
+
+#include "particles/ImpactXParticleContainer.H"
+#include "mixin/beamoptic.H"
+#include "mixin/beamoptic.H"
+
+#include <AMReX_Extension.H>
+#include <AMReX_REAL.H>
+
+#include <cmath>
+
+
+namespace impactx
+{
+    struct Sol
+    : public elements::BeamOptic<Sol>,
+      public elements::Thick
+    {
+        static constexpr auto name = "Sol";
```

Comment:
Is this the Mad-X equivalent name or shall we call it `Solenoid` or `IdealSolenoid`?

Reply:
@cemitch99 on Slack:
> MAD-X by default has only an ideal solenoid element, denoted by SOLENOID (Section 10.9, p. 78 of the user guide).
>
> It is possible to access some features of the tracking code PTC within MAD-X, which probably allows for a soft-edge model, but PTC is really a separate code, and not standard in MAD.  A soft-edge solenoid appears in MaryLie, IMPACT-Z, and MaryLie/IMPACT.